File: ios/ReactNativeEventEmitter.m
//
//  ReactNativeEventEmitter.m
//  plottr_mobile
//
//  Created by Cameron Sutter on 5/14/20.
//
// See: http://reactnative.dev/docs/native-modules-ios.html#exporting-swift
//

#import <Foundation/Foundation.h>

#import <React/RCTBridge.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(ReactNativeEventEmitter, RCTEventEmitter)

RCT_EXTERN_METHOD(supportedEvents)

@end

File: ios/ReactNativeEventEmitter.swift
//
//  ReactNativeEventEmitter.swift
//  plottr_mobile
//
//  Created by Cameron Sutter on 5/14/20.
//

import Foundation
//import React

@objc(ReactNativeEventEmitter)
open class ReactNativeEventEmitter: RCTEventEmitter {

  override init() {
    super.init()
    EventEmitter.sharedInstance.registerEventEmitter(eventEmitter: self)
  }

  /// Base overide for RCTEventEmitter.
  ///
  /// - Returns: all supported events
  @objc open override func supportedEvents() -> [String] {
    return EventEmitter.sharedInstance.allEvents
  }

}
